Claudia/src/main/java/ReferenceData.java

```diff
@@ -0,0 +1,20 @@
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ReferenceData {
+
+    public static boolean isValidDate(String DateEx)
+    {
+        String regex = "^(((19[0-9]{2})|(2[01][0-9]{2}))-(1[02]|0[13578])-(31))|(((19[0-9]{2})|(2[01][0-9]{2}))-(1[0-2]|0[1,3-9])-(29|30))|(((19[0-9]{2})|(2[01][0-9]{2}))-(1[0-2]|0[1-9])-(0[1-9]|1[0-9]|2[0-8]))$";
```

Comment:
Good job! You have covered a lot of cases!
The only thing that you can improve is to move the year part out of brackets (the year part is the same across all the three cases)

Reply:
Done